common/src/main/java/org/opensearch/sdk/SdkClient.java

```diff
@@ -46,7 +46,12 @@ default PutDataObjectResponse putDataObject(PutDataObjectRequest request) {
             if (cause instanceof InterruptedException) {
                 Thread.currentThread().interrupt();
             }
-            throw new OpenSearchException(cause);
+            // Rethrow unchecked Exceptions
+            if (cause instanceof RuntimeException) {
+                throw (RuntimeException) cause;
+            } else {
+                throw new OpenSearchException(cause);                
+            }
```

Comment:
Given that CompletionException extends RunTimeException do we except non-RunTimeException?

Reply:
Do we "expect"?  No.  But I spent quite some time debugging a mysterious NullPointerException which was swallowed without handling like this.

Reply:
Also, I've already extracted the cause from the Completion Exception:
```java
Throwable cause = e.getCause();
```